app/models/framework/definition/ast/semantic_checker.rb

```diff
@@ -77,12 +77,24 @@ def raise_when_lookup_reference_invalid(field)
         end
 
         def raise_when_management_field_invalid(info)
-          management_charge_modifier = [info.dig(:column_based, :column_names), info.dig(:flat_rate, :column)].compact.first(&:present?)
-          return if management_charge_modifier.blank?
+          of_columns_for(info).each { |referenced_field_name| validate_management_field(referenced_field_name) }
+        end
 
-          Array(management_charge_modifier).each do |referenced_field_name|
-            validate_management_field(referenced_field_name)
-          end
+        def of_columns_for(info)
+          Array(
+            if info[:column_based]
+              varies_by = Array(info[:column_based][:column_names])
+
+              percentage_columns = info[:column_based][:value_to_percentage]
+              percentage_columns = percentage_columns.values
+                                     .select { |percentage_details| percentage_details[:column].present? }
+                                     .map    { |percentage_details| percentage_details[:column] }
+
+              varies_by.concat percentage_columns
+            elsif info[:flat_rate]
+              info.dig(:flat_rate, :column)
+            end
```

Comment:
Is it worth catching anything that neither satisfied the `if` or the `elsif` here?

Reply:
Well there's `info[:sector_based]` but fundamentally these are checks. The outer `Array` cast will send `Array[nil] => []` in that case, and no check will fail, which is what we want.

Reply:
Though if you'd prefer an `else []` that would certainly be explicit?